src/dataflow/src/render/sources.rs

```diff
@@ -816,6 +862,148 @@ where
     sealed_stream
 }
 
+/// A `PersistedSourceManager` stores maps from global identifiers to the one rendered persistent
+/// instance of a source.
+///
+/// NOTE: There can only ever be one running (rendered) instance of a persisted source. The manager
+/// will panic when trying to add more than on instance for a given source id.
+pub struct PersistedSourceManager {
+    /// Handles that allow setting the compaction frontier for a persisted source.
+    pub compaction_handles: HashMap<GlobalId, Weak<Box<dyn CompactionHandle>>>,
```

Comment:
do we need the CompactionHandle trait? I have a mild preference for this simply being
```
HashMap<GlobalId, BoundedCompactionHandle>
```

Reply:
Will do! This is a remnant from when I was storing a `Box<dyn CompactionHandle>` in rendering state instead of the shinier and better `PersistedSourceManager`, which hides these details internally.